app/controllers/carto/api/received_notifications_controller.rb

```
@@ -0,0 +1,41 @@
+# encoding: utf-8
+
+module Carto
+  module Api
+    class ReceivedNotificationsController < ::Api::ApplicationController
+      include Carto::ControllerHelper
+      extend Carto::DefaultRescueFroms
+
+      ssl_required :update
+
+      before_filter :load_user
+      before_filter :load_notification, only: [:update]
+
+      setup_default_rescues
+
+      def update
+        changed_notification = params[:notification]
+        if changed_notification
+          read_at = changed_notification[:read_at]
+          @received_notification.read_at = DateTime.parse(read_at) if read_at
+          @received_notification.save!
+        end
+
+        render_jsonp ReceivedNotificationPresenter.new(@received_notification)
+      rescue ArgumentError
+        render_jsonp({ errors: { read_at: 'invalid date format' } }, 422)
```

I'd log the exception, maybe warn or debug level. The assumption of the wrong date is not bad, but might hide other errors.